users/tests/test_user_login.py

```diff
@@ -22,3 +33,20 @@ def test_user_client_should_not_log_in_when_email_is_none(self):
 
     def test_user_client_should_not_log_in_when_password_is_none(self):
         self.assertFalse(self.client.login(email="testuser@codepoets.it", password=None))
+
+    def test_user_should_redirect_to_change_password_form_if_need_change_password_is_true(self):
+        data = {"username": self.user.email, "password": self.password}
+        response = self.client.post(reverse("login"), data=data)
+        self.assertRedirects(response, reverse("password_change_on_login"), status_code=302, target_status_code=200)
+
+    def test_user_should_redirect_to_home_if_need_change_password_is_false(self):
+        data = {"username": self.user.email, "password": self.password}
+        self.user.need_change_password = False
+        self.user.save()
```

Comment:
This can be withdraw to separate private method and used only here. Methods have advantage that with proper method name you can easily says what method is doing. It simplify code and make it easier to read. Please split this code to separate private method,

Reply:
Done.